Feat/provider registration

Harr-izle opened the pull request.

Harr-izle committed:
feat:validates payment method fields,styles the service preference page and the preview page

Harr-izle committed:
I want to get the current changes from dev Merge branch 'dev' of https://github.com/AmaliTech-Training-Academy/proximity-finder-FE into feat/provider-registration

Harr-izle committed:
I want to get the current changes in dev Merge branch 'dev' of https://github.com/AmaliTech-Training-Academy/proximity-finder-FE into feat/provider-registration

Harr-izle committed:
feat:updates UI design for basic info display

Harr-izle committed:
feat:fixes error message display

Harr-izle committed:
want to get the current changes from devMerge branch 'dev' of https://github.com/AmaliTech-Training-Academy/proximity-finder-FE into feat/provider-registration

Harr-izle committed:
feat:creates resuable component for addition of social media links

Harr-izle committed:
feat:adds social media icons to assets

Harr-izle committed:
feat:adds missing fields to the payment methods page

Harr-izle committed:
I want to get current changes from dev Merge branch 'dev' of https://github.com/AmaliTech-Training-Academy/proximity-finder-FE into feat/provider-registration

Harr-izle committed:
feat:validates payment option form

Harr-izle committed:
feat:validates and adds remaining styles to about business page

ernestbrayce commented on src/app/features/pro-registration/components/profile-update/profile-update.component.ts:
```diff
@@ -22,4 +28,19 @@ export class ProfileUpdateComponent {
+  }
+
+  onFileSelected(event: any): void {
+    const file = event.target.files[0];
```
Add null safety here

`if (input?.files?.length) {`

ernestbrayce commented on src/app/features/pro-registration/components/profile-update/profile-update.component.ts:
```diff
@@ -22,4 +28,19 @@ export class ProfileUpdateComponent {
+    document.getElementById('file')?.click();
+  }
+
+  onFileSelected(event: any): void {
```
Stop using type `any`

ernestbrayce reviewed (approved):
Fix the comments and merge

BlissArmah-Nwanwah commented on src/app/features/pro-registration/components/payment-method/payment-method.component.ts:
```diff
@@ -10,22 +12,37 @@ interface paymentMethod {
 })
 export class PaymentMethodComponent {
-  paymentMethod!: paymentMethod[];
+  paymentMethod: paymentMethod [] =[
+    {name: 'Mobile Money'},
+    {name: 'Bank Account'},
+    {name: 'Paypal Account'},
+  ]
```
```suggestion
  paymentMethod =[
    {name: 'Mobile Money'},
    {name: 'Bank Account'},
    {name: 'Paypal Account'},
  ]
```

BlissArmah-Nwanwah replied:
Typescript will automatically infer the type from the value assigned to the variable (Payment Method) so there is no need to explicitly type it 

```suggestion
  paymentMethod =[
    {name: 'Mobile Money'},
    {name: 'Bank Account'},
    {name: 'Paypal Account'},
  ]
```


Also you can create a constant for your values and use them here

BlissArmah-Nwanwah committed:
Update src/app/features/pro-registration/components/payment-method/payment-method.component.ts

Typescript will infer the assigned valued by default so there is no need to explicitly type paymentMethod

Harr-izle committed:
feat:resolves conflict

Harr-izle committed:
fix:fixes PR comments

Harr-izle committed:
Merge branch 'feat/provider-registration' of https://github.com/AmaliTech-Training-Academy/proximity-finder-FE into feat/provider-registration
